docs/configuring.md

```diff
@@ -0,0 +1,285 @@
+
+- On AIX systems, the kernel dynamically adjusts the shared memory settings as required. No special configuration is required.
+- On Linux systems, the **SHMMAX** setting limits the amount of shared memory that can be allocated, which affects the shared classes cache size. You can find the value of **SHMMAX** for your system in the `/proc/sys/kernel/shmmax` file. For non-persistent caches, set this value to an appropriate size for your applications. To make these changes permanent, edit `/etc/sysctl.conf` and reboot your system.
+- On macOS systems, you must set **kern.sysv.shmmax** and **kern.sysv.shmall** when using a nonpersistent cache. Modify the settings in your `/etc/sysctl.conf` file and reboot your system. To check the value, run `sysctl kern.sysv.shmmax`.
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** The virtual address space of a process is shared between the shared classes cache and the Java heap. Increasing the maximum size for the shared classes cache might reduce the size of the Java heap that you can create.
+
+Shared memory limits are also important when configuring large page memory allocation on Linux systems. For more information, see [Configuring large page memory allocation: Linux systems](#linux-systems).
+
+## Setting resource limits (z/OS)
+
+Resource limits imposed by z/OS might affect Java operations. To learn how these resource limits are set, see [Customizing the BPXPRMxx member of SYS1.PARMLIB](https://www.ibm.com/support/knowledgecenter/SSLTBW_2.4.0/com.ibm.zos.v2r4.bpxb200/cusis.htm).
+
+The OpenJ9 class data sharing feature is implemented by using shared memory segments on z/OS. Special consideration should be given to the following parameters that relate to the shared memory and IPC semaphore settings:
+
+<!-- - **MAXSHAREPAGES** (This no longer seems to do anything, per https://www.ibm.com/support/knowledgecenter/SSLTBW_2.4.0/com.ibm.zos.v2r4.ieae200/spb.htm). Also removed from table below. -->
+- **IPCSHMSPAGES**
+- **IPCSHMMPAGES**
+- **IPCSHMNSEGS**
+
+Incorrect or suboptimal settings might prevent shared classes from working or impact performance. By default, the VM attempts to create a 16 MB cache. If you set a cache size for your application by specifying the [`-Xscmx`](xscmx.md) option on the command line, the VM rounds the value up to the nearest megabyte. Ensure that the value set for **IPCSHMMPAGES** takes this adjustment into consideration.
+
+To see the current settings, enter the following z/OS operator command:
```

Comment:
I'd suggest this paragraph on how to see current OMVS settings, and the table below with suggested min values should be bumped up and placed right after the initial paragraph of this section.   i.e. placed ahead of the shared memory options above.